develop/core/views.py

```diff
@@ -22,26 +27,29 @@ class ProductAccessView(ProductRequiredMixin, TemplateView):
 
     def get(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
-        context['object'] = self.model.on_site.get(slug=kwargs['slug'])
+        site = get_current_site(request)
```

Comment:
I'd use the same pattern from line 19 since a middleware should already attach a site to the request object.  Do a check for the attribute on the request object first and then use get_current_site() if not.

Reply:
Added SetSiteToRequestMixin